File: leetcode/简单题/24 两两交换链表中的节点/another.c
struct ListNode* swapPairs(struct ListNode* head){
    if(head == NULL || head->next == NULL)
            return head;
    struct ListNode* node = malloc(sizeof(struct ListNode));
    node->next = head;
    struct ListNode* temp = node;
    while(node->next && node->next->next){
        struct ListNode* swap = node->next;
        node->next = node->next->next;
        swap->next = swap->next->next;
        node->next->next = swap;
        node = node->next->next;
    }
    return temp->next;
}

//差点绕晕了……链表题一定要自己多画图画图！否则就会和我一样有丢失节点的现象发生，画个图什么都清楚了

File: leetcode/简单题/24 两两交换链表中的节点/another.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
	ans := &ListNode{-1, head}
	res := ans
	for ans.Next != nil && ans.Next.Next != nil {
		right1 := ans.Next
		right2 := right1.Next
		right3 := right2.Next
		ans.Next = right2
		right2.Next = right1
		right1.Next = right3
		ans = ans.Next.Next
	}
	return res.Next
}

//hhhhhh跟上次一样，不多说了，就是画个图什么都清楚了，right1 right2 right3的，就是交换一个链表里的两个元素，要涉及四个节点
//最前面的用ans了，后面三个用right1 right2 right3来表示了，比用next.next之类的要清楚一点
